src/kotoba/agent/cli.c

```diff
@@ -0,0 +1,117 @@
+/*
+ Author: Harshit Gandhi
+ SPDX-FileCopyrightText: © 2025 Harshit Gandhi <gandhiharshit716@gmail.com>
+
+ SPDX-License-Identifier: GPL-2.0-only
+*/
+
+#include "cli.h"
+#include "file_operations.h"
+#include "database.h"
+#include "match.h"
+
+MatchCallbacks cliCallbacks =
+  { .onNo = cli_onNoMatch,
+    .onFull = cli_onFullMatch,
+    .onBeginOutput = cli_onBeginOutput,
+    .onBetweenIndividualOutputs = cli_onBetweenIndividualOutputs,
+    .onEndOutput = cli_onEndOutput,
+    .onDiff = cli_onDiff
+  };
+
+int matchCliFileWithLicenses(KotobaState* state, const Licenses* licenses, int argi, char** argv) {
+  File file;
+  file.id = argi;
+  file.fileName = argv[argi];
+  if (!readTokensFromFile(file.fileName, &(file.tokens), DELIMITERS))
+    return 0;
+
+  int result = matchFileWithLicenses(state, &file, licenses, &cliCallbacks);
+
+  tokens_free(file.tokens);
+
+  return result;
+}
+
+int handleCliMode(KotobaState* state, const Licenses* licenses, int argc, char** argv, int fileOptInd) {
+#ifdef KOTOBA_MULTI_THREAD
+  #pragma omp parallel
+#endif
+  {
+    KotobaState threadLocalStateStore = *state;
+    KotobaState* threadLocalState = &threadLocalStateStore;
+
+#ifdef KOTOBA_MULTI_THREAD
+    #pragma omp for schedule(dynamic)
+#endif
+    for (int fileId = fileOptInd; fileId < argc; fileId++) {
+      matchCliFileWithLicenses(threadLocalState, licenses, fileId, argv);
+    }
+  }
+
+  return 1;
+}
+
+int cli_onNoMatch(KotobaState* state, const File* file) {
```

Comment:
Same here? Shouldnt be reusing the functions or portion of code instead of writing exact same stuff???